source/Planet.h

```diff
@@ -178,11 +181,12 @@ class Planet {
 	std::set<std::string> attributes;
 	
 	std::set<const Sale<Ship> *> shipSales;
-	std::set<const Sale<Outfit> *> outfitSales;
+	std::set<const OutfitSale *> outfitSales;
 	// The lists above will be converted into actual ship lists when they are
 	// first asked for:
 	mutable Sale<Ship> shipyard;
-	mutable Sale<Outfit> outfitter;
+	mutable Sale<Outfit> outfits;
+	mutable OutfitSale outfitSale;
```

Comment:
Why have two? One `OutfitSale` should be enough.

Reply:
just look at the outfits function, it returns outfitsale in the form of sale<outfits>, that is needed for some classes that do like set comparison

Reply:
Yeah but a map is a set just with a few more things, so you can modify those functions to use the map.

Reply:
I really don't know how to do it though never used a map before this project

Reply:
A map as the almost the same interface as a set, so it should work without a lot of additional changes

Reply:
rewriting the function is fine then ? like doing my own set_intersection ?

Reply:
What set_intersection?

Reply:
LocationFilter it was SetsIntersect actually I defined a new overload for the map

Reply:
`outfitSale` => `outfitter` please.